### ISSUE TITLE
[UI/UX] APIM 3 General Accessibility

### ISSUE THREAD

**Issue by tmkasun** [wso2/product-apim#3935]
**Description:**
We need to consider the accessibility best practices mentioned in 

http://www.w3.org/TR/wai-aria-practices/

when developing web apps for APIM products.

Since we are using Material UI for building the React SPA apps, [MUI already supports](https://github.com/mui-org/material-ui/issues/61) ARIA practices and enforce it in some components.

Further, we need to follow the guidelines given in the [React Documentation](https://reactjs.org/docs/accessibility.html) for adapting the aria practices.

# WCAG21 reference

- https://www.w3.org/WAI/WCAG21/quickref/?currentsidebar=%23col_overview&levels=aaa

**Affected Product Version:**

- APIM 3.x

**Comment by tmkasun** [comment]
We could use [react-a11y](https://github.com/reactjs/react-a11y) to identify the missing ARIA practices at the development time.

We could use [Google LightHouse](https://developers.google.com/web/tools/lighthouse/) command line tool and integrate it with the product build to get the accessibility coverage report or we could use @dakshika's [pa11y-full-report](https://github.com/dakshika/pa11y-full-report) tool to get a more descriptive report

**Comment by tmkasun** [comment]
For the comparison, 

APIM 3.x accessibility report from Google LightHouse

![image](https://user-images.githubusercontent.com/3313885/52053758-411d6d80-2580-11e9-9d84-2b4ea1eb8339.png)

accessibility report for latest WUM updated APIM 2.7 

![image](https://user-images.githubusercontent.com/3313885/52053761-44185e00-2580-11e9-9362-ea40bc1bd5ea.png)

**Comment by tmkasun** [comment]
Using react-axe [article](https://web.dev/accessibility-auditing-react/)

**Comment by tmkasun** [comment]
Automating lighthouse report with puppeteer: [Sample repo](https://github.com/addyosmani/puppeteer-webperf#generate-a-lighthouse-report)

**Comment by erangatl** [comment]
Created the following issues to track
#8390
#8386
#8383
#8380
#8379
#8374
#8370
#8342
#8143
#8129
#8102
#8101
#8086
#7988

**Comment by erangatl** [comment]
Following doc is created as a guideline 
https://docs.google.com/document/d/10Y-vuV458Q70Qo9N-RN5QXegbOeDl_Ox-7ZTBcO4b4I/edit?usp=sharing

**Comment by tmkasun** [comment]
@erangatl , Above doc, is not publicly accessible? Shall we add it to [wikis](https://github.com/wso2/carbon-apimgt/wiki) ?

**Comment by erangatl** [comment]
> @erangatl , Above doc, is not publicly accessible? Shall we add it to [wikis](https://github.com/wso2/carbon-apimgt/wiki) ?

The plan was to update the HIG, shall we update all the design-related stuff in HIG?

**Comment by erangatl** [comment]
React A11y library is deprecated.

**Comment by erangatl** [comment]
axe - Web Accessibility Testing
https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd

![image](https://user-images.githubusercontent.com/20179540/88824132-92450680-d1e3-11ea-8233-815dbf0b9ef1.png)

**Comment by erangatl** [comment]
Jest axe [article](https://medium.com/hackernoon/testing-react-with-jest-axe-and-react-testing-library-accessibility-34b952240f53)
[jest-axe](https://github.com/nickcolley/jest-axe)

**Comment by rmsamitha** [comment]
Closing the issue as this is older than 3 years. Please reopen if the issue is valid still.
